go/mcap/reader.go

```diff
@@ -95,38 +103,115 @@ func (r *Reader) unindexedIterator(topics []string, start uint64, end uint64) *u
 	}
 }
 
-func (r *Reader) indexedMessageIterator(topics []string, start uint64, end uint64) *indexedMessageIterator {
+func (r *Reader) indexedMessageIterator(
+	topics []string,
+	start uint64,
+	end uint64,
+	order ReadOrder,
+) *indexedMessageIterator {
 	topicMap := make(map[string]bool)
 	for _, topic := range topics {
 		topicMap[topic] = true
 	}
 	r.l.emitChunks = true
 	return &indexedMessageIterator{
-		lexer:    r.l,
-		rs:       r.rs,
-		channels: make(map[uint16]*Channel),
-		schemas:  make(map[uint16]*Schema),
-		topics:   topicMap,
-		start:    start,
-		end:      end,
+		lexer:     r.l,
+		rs:        r.rs,
+		channels:  make(map[uint16]*Channel),
+		schemas:   make(map[uint16]*Schema),
+		topics:    topicMap,
+		start:     start,
+		end:       end,
+		indexHeap: rangeIndexHeap{order: order},
+	}
+}
+
+type readOptions struct {
+	start    int64
+	end      int64
+	topics   []string
+	useIndex bool
+	order    ReadOrder
+}
+
+func defaultReadOptions() readOptions {
+	return readOptions{
+		start:    0,
+		end:      math.MaxInt64,
+		topics:   nil,
+		useIndex: true,
+		order:    ReadOrderFile,
+	}
+}
+
+type ReadOpt func(*readOptions) error
+
+func ReadMessagesAfter(start int64) ReadOpt {
```

Comment:
since these are supplied to a call to Messages, I wonder if `After`, `Before`, `WithTopics`, `WithIndex` (or `UsingIndex` I guess) might be nicer shortenings without any loss of information

Reply:
It's just a matter of cluttering the `mcap` module namespace. I'll see what I can do if I make a separate sub-package.